### PR TITLE
repository -> Definitions in IO manager docstrings

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

**Review by schrockn — APPROVED** [review]

File: python_modules/dagster/dagster/_core/storage/fs_io_manager.py [hunk 1]
```diff
@@ -62,14 +62,12 @@ def asset1():
...
-            return with_resources(
-                [asset1, asset2],
-                resource_defs={
-                    "io_manager": fs_io_manager.configured({"base_dir": "/my/base/path"})
-                },
-            )
+        defs = Definitions(
+            assets=[asset1, asset2],
+            resources={
+                "io_manager": fs_io_manager.configured({"base_dir": "/my/base/path"})
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/storage/fs_io_manager.py:68]: Actually are new Pythonic variants of these avaiable? cc: @jamiedemaria 

Also remove unused imports (e.g. with_resources)

**Review by schrockn — CHANGES_REQUESTED** [review]
spoke too soon

**Review by schrockn — COMMENTED** [review]
make sure to check all the imports in the docstrings

File: python_modules/dagster/dagster/_core/storage/fs_io_manager.py (hunk 1 above)
Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/storage/fs_io_manager.py:68]: I dont think the three io managers in this pr have been updated yet

Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/storage/fs_io_manager.py:68]: although working on gcs now https://github.com/dagster-io/dagster/pull/13653

**Comment by sryza** [comment]
@schrockn thanks for catching - I just updated the PR

**Review by schrockn — APPROVED** [review]

**Pull request merged**